test/e2e_tests/specs/Block/block.spec.ts

```diff
@@ -0,0 +1,344 @@
+        {tag: ['@TC-144', '@regression']},
+        async ({pageManager: userAPageManager, userBPageManager}) => {
+          const {pages: userBPages} = userBPageManager.webapp;
+
+          // Step 1: User A sends connection request to User B
+          await test.step('Preconditions: User A connects with User B', async () => {
+            await connectUsersManually(userA, userB, userAPageManager, userBPageManager, false);
+          });
+
+          // Step 2: User A blocks User B from conversation list
+          await blockUserFromConversationList(userAPageManager, userB);
+          // Step 3: User B does not see Connection Request from User A
+          expect(userBPages.conversationList()).not.toContain(userA.fullName);
+        },
+      );
+    });
+
+    test(
+      'Verify you can unblock someone from conversation list options',
+      {tag: ['@TC-148', '@regression']},
+      async ({pageManager: userAPageManager, userBPageManager}) => {
+        test.slow();
+        const {pages: userAPages, modals: userAModals, components: userAComponents} = userAPageManager.webapp;
+        const {pages: userBPages} = userBPageManager.webapp;
+
+        await test.step('User A blocks User B', async () => {
+          await blockUserFromConversationList(userAPageManager, userB, {handleUnableToOpenModal: true});
+        });
+
+        await test.step('User A unblocks User B from Search List', async () => {
+          await userAComponents.conversationSidebar().clickConnectButton();
+          await userAPages.startUI().searchInput.fill(userB.username);
+          await userAPages.startUI().selectUser(userB.username);
+          await userAModals.userProfile().unblockButton.click();
+          await userAModals.unblockWarningModal().unblockButton.click();
+        });
+
+        await test.step('User B receives message sent by User A', async () => {
+          await userAPages.conversationList().openConversation(userB.fullName);
+          await userAPages.conversation().sendMessage('Message after unblocking');
+          await userBPages.conversationList().openConversation(userA.fullName);
+          expect(await userBPages.conversation().messageCount()).toBe(1);
+        });
+
+        await test.step('User A receives message sent by User B', async () => {
+          await userBPages.conversationList().openConversation(userA.fullName);
+          await userBPages.conversation().sendMessage('Message after being unblocked');
+          await userAPages.conversationList().openConversation(userB.fullName);
+          expect(await userBPages.conversation().messageCount()).toBe(2);
+        });
+      },
+    );
+  });
+
+  test.describe('Block: User A and User B are in the same team', () => {
+    test.beforeEach('Team Test', async ({api, pageManager, browser}) => {
+      userA = getUser();
+      userB = getUser();
+      const teamName = 'Team Name';
+      const context = await browser.newContext();
+      const page = await context.newPage();
```

Comment:
Is it really necessary to open a new page just to log in userB and then do nothing with it? I was able to remove the whole context without the test failing.